### PR TITLE
Change to use F-string

### PR DESCRIPTION
**Pull request by ZwwWayne**
After discussion, MMDet V2.0 will no longer support python 3.5.
Therefore, to reduce the burden of `.format`, this PR change all the strings to `f'{}'`, which is simpler and shorter without losing clarity.

### CODE REVIEW

File: mmdet/core/anchor/anchor_generator.py [hunk 1]
```diff
@@ -64,21 +64,21 @@ def __init__(self,
         # check center and center_offset
         if center_offset != 0:
             assert centers is None, 'center cannot be set when center_offset' \
-                '!=0, {} is given.'.format(centers)
+                '!=0, {centers} is given.'
```
Comment (hellock) [inline mmdet/core/anchor/anchor_generator.py:67]: Missing the leading `f`.

File: mmdet/core/anchor/anchor_generator.py [hunk 2]
```diff
@@ -259,21 +259,18 @@ def single_level_valid_flags(self,
...
-        repr_str += '{}scale_major={},\n'.format(indent_str, self.scale_major)
-        repr_str += '{}octave_base_scale={},\n'.format(indent_str,
-                                                       self.octave_base_scale)
-        repr_str += '{}scales_per_octave={},\n'.format(indent_str,
-                                                       self.scales_per_octave)
-        repr_str += '{}num_levels={}\n'.format(indent_str, self.num_levels)
-        repr_str += '{}centers={},\n'.format(indent_str, self.centers)
-        repr_str += '{}center_offset={})'.format(indent_str,
-                                                 self.center_offset)
+        indent_str = '\n    '
```
Comment (hellock) [inline mmdet/core/anchor/anchor_generator.py:262]: The name is misleading. The following may be better.
```python
indent_str = '    '
repr_str = self.__class__.__name__ + '(\n'
repr_str += f'{indent_str}strides={self.strides},\n'
```

File: mmdet/core/bbox/samplers/sampling_result.py [hunk 3]
```diff
@@ -71,7 +71,7 @@ def __nice__(self):
         data = self.info.copy()
         data['pos_bboxes'] = data.pop('pos_bboxes').shape
         data['neg_bboxes'] = data.pop('neg_bboxes').shape
-        parts = ['\'{}\': {!r}'.format(k, v) for k, v in sorted(data.items())]
+        parts = [f'\'{k}\': {v!r}' for k, v in sorted(data.items())]
```
Comment (hellock) [inline mmdet/core/bbox/samplers/sampling_result.py:74]: `f"'{k}': {v!r}"`

File: mmdet/datasets/pipelines/compose.py [hunk 4]
```diff
@@ -31,6 +31,6 @@ def __repr__(self):
         format_string = self.__class__.__name__ + '('
         for t in self.transforms:
             format_string += '\n'
-            format_string += '    {0}'.format(t)
+            format_string += f'    {t:0}'
```
Comment (hellock) [inline mmdet/datasets/pipelines/compose.py:34]: `f'    {t}'`

File: mmdet/datasets/pipelines/formating.py [hunk 5]
```diff
@@ -75,8 +74,8 @@ def __call__(self, results):
         return results
 
     def __repr__(self):
-        return self.__class__.__name__ + '(keys={}, order={})'.format(
-            self.keys, self.order)
+        return self.__class__.__name__ + \
+            f'(keys={self.keys}, order={self.order})'
```
Comment (hellock) [inline mmdet/datasets/pipelines/formating.py:78]: Check if the break line is necessary.

File: mmdet/datasets/pipelines/transforms.py [hunk 6]
```diff
@@ -222,8 +222,7 @@ def bbox_flip(self, bboxes, img_shape, direction):
             flipped[..., 1::4] = h - bboxes[..., 3::4]
             flipped[..., 3::4] = h - bboxes[..., 1::4]
         else:
-            raise ValueError(
-                'Invalid flipping direction "{}"'.format(direction))
+            raise ValueError(f'Invalid flipping direction "{direction}"')
```
Comment (hellock) [inline mmdet/datasets/pipelines/transforms.py:225]: Use single quote to wrap the str.

File: mmdet/datasets/pipelines/transforms.py [hunk 7]
```diff
@@ -511,12 +507,12 @@ def __call__(self, results):
...
-                         self.brightness_delta,
-                         (self.contrast_lower, self.contrast_upper),
-                         (self.saturation_lower, self.saturation_upper),
-                         self.hue_delta)
+        repr_str += f'(\nbrightness_delta={self.brightness_delta},\n'
+        repr_str += f'contrast_range='
+        repr_str += f'{(self.contrast_lower, self.contrast_upper)},\n'
+        repr_str += f'saturation_range='
+        repr_str += f'{(self.saturation_lower, self.saturation_upper)},\n'
+        repr_str += f'hue_delta={self.hue_delta})'
```
Comment (hellock) [inline mmdet/datasets/pipelines/transforms.py:515]: The formatting is different from others.

File: mmdet/datasets/pipelines/transforms.py [hunk 8]
```diff
@@ -867,5 +861,5 @@ def __call__(self, results):
 
     def __repr__(self):
         repr_str = self.__class__.__name__
-        repr_str += '(transforms={})'.format(self.transforms)
+        repr_str += f'(transforms={self.transforms})'
```
Comment (hellock) [inline mmdet/datasets/pipelines/transforms.py:864]: `repr_str = self.__class__.__name__ + f'(transforms={self.transforms})'`

File: mmdet/models/backbones/hrnet.py [hunk 9]
```diff
@@ -43,18 +43,18 @@ def __init__(self,
     def _check_branches(self, num_branches, num_blocks, in_channels,
                         num_channels):
         if num_branches != len(num_blocks):
-            error_msg = 'NUM_BRANCHES({}) <> NUM_BLOCKS({})'.format(
-                num_branches, len(num_blocks))
+            error_msg = f'NUM_BRANCHES({num_branches}) ' \
+                f'<> NUM_BLOCKS({len(num_blocks)})'
```
Comment (hellock) [inline mmdet/models/backbones/hrnet.py:47]: `<>` -> `!=`

File: mmdet/models/necks/fpn.py [hunk 10]
```diff
@@ -45,7 +45,7 @@ class FPN(nn.Module):
         >>> self = FPN(in_channels, 11, len(in_channels)).eval()
         >>> outputs = self.forward(inputs)
         >>> for i in range(len(outputs)):
-        ...     print('outputs[{}].shape = {!r}'.format(i, outputs[i].shape))
+        ...     print(f'outputs[{i}].shape = {outputs[i].shape!r}')
```
Comment (hellock) [inline mmdet/models/necks/fpn.py:48]: The `!r` is unnecessary.

File: mmdet/ops/carafe/grad_check.py [hunk 11]
```diff
@@ -39,9 +39,9 @@
     torch.cuda.synchronize()
     time_backward += timer.since_last_check()
     bar.update()
-print('\nCARAFE time forward: {} ms/iter | time backward: {} ms/iter'.format(
-    (time_forward + 1e-3) * 1e3 / loop_num,
-    (time_backward + 1e-3) * 1e3 / loop_num))
+print(f'\nCARAFE time forward: {(time_forward + 1e-3) * 1e3 / loop_num} '
+      f'ms/iter | time backward: {(time_backward + 1e-3) * 1e3 / loop_num}'
+      ' ms/iter')
```
Comment (hellock) [inline mmdet/ops/carafe/grad_check.py:44]: Compute `(time_forward + 1e-3) * 1e3 / loop_num` ahead.

File: mmdet/ops/dcn/deform_conv.py [hunk 12]
```diff
@@ -205,11 +203,10 @@ def __init__(self,
 
         assert not bias
         assert in_channels % groups == 0, \
-            'in_channels {} cannot be divisible by groups {}'.format(
-                in_channels, groups)
+            f'in_channels {in_channels} cannot be divisible by groups {groups}'
         assert out_channels % groups == 0, \
-            'out_channels {} cannot be divisible by groups {}'.format(
-                out_channels, groups)
+            f'out_channels {out_channels} cannot be divisible ' \
```
Comment (hellock) [inline mmdet/ops/dcn/deform_conv.py:208]: `cannot be` -> `is not`

File: mmdet/ops/roi_pool/roi_pool.py [hunk 13]
```diff
@@ -69,7 +69,7 @@ def forward(self, features, rois):
 
     def __repr__(self):
         format_str = self.__class__.__name__
-        format_str += '(out_size={}, spatial_scale={}'.format(
-            self.out_size, self.spatial_scale)
-        format_str += ', use_torchvision={})'.format(self.use_torchvision)
+        format_str += f'(out_size={self.out_size}, '
+        format_str += f'spatial_scale={self.spatial_scale}'
+        format_str += f', use_torchvision={self.use_torchvision})'
```
Comment (hellock) [inline mmdet/ops/roi_pool/roi_pool.py:74]:
```python
format_str += f'spatial_scale={self.spatial_scale}, '
format_str += f'use_torchvision={self.use_torchvision})'
```

File: tests/test_config.py [hunk 14]
```diff
@@ -29,14 +29,14 @@ def test_config_build_detector():
     from mmdet.models import build_detector
 
     config_dpath = _get_config_directory()
-    print('Found config_dpath = {!r}'.format(config_dpath))
+    print(f'Found config_dpath = {config_dpath!r}')
```
Comment (hellock) [inline tests/test_config.py:32]: The `!r` is unnecessary.

File: mmdet/datasets/pipelines/formating.py (hunk 5 above)
Comment (ZwwWayne) [reply mmdet/datasets/pipelines/formating.py:78]: Putting them in on line will have max length > 79.

**Pull request merged**